easybatch-core/src/main/java/org/easybatch/core/mapper/ObjectMapper.java

```diff
@@ -158,8 +174,12 @@ private T createInstance() throws RecordMappingException {
 
     private void convertValue(T result, String field, String value, Method setter, Class<?> type, TypeConverter typeConverter) throws RecordMappingException {
         try {
-            Object typedValue = typeConverter.convert(value);
-            setter.invoke(result, typedValue);
+            if (emptyToNull && value.isEmpty()) {
+                setter.invoke(result, new Object[]{null});
```

Comment:
This is fine for Object types, but invoking a setter with null value on a primitive type cause a IllegalArgumentException. Here is a test case:

```java
    @Test
    public void whenAttemptingToSetAnEmptyValueToAPrimitiveType_ThenShouldSetDefaultValueIfOptionEnabled() throws Exception {

        ObjectMapper<Person> mapper = new ObjectMapper<Person>(Person.class, true);

        Map<String, String> values = new HashMap<String, String>();
        values.put("age", "");

        Person person = mapper.mapObject(values);
        assertThat(person).isNotNull();
        assertThat(person.getAge()).isEqualTo(0);
    }
```